File: app/src/main/java/daocacao/myprecious/data/Day.java
package daocacao.myprecious.data;

import io.realm.RealmList;
import io.realm.RealmObject;
import io.realm.annotations.PrimaryKey;

public class Day extends RealmObject {

    @PrimaryKey
    private String date;
    private int month;
    private int year;
    private RealmList<Gold> golds;

    public Day() {
    }

    public Day(String date) {
        this.date = date;
        month = Integer.parseInt(date.split("\\.")[1]);
        year = Integer.parseInt(date.split("\\.")[2]);
        golds = new RealmList<>();
    }

    public RealmList<Gold> getGolds() {
        return golds;
    }
}

File: app/src/main/java/daocacao/myprecious/data/Day.kt
package daocacao.myprecious.data

import io.realm.RealmList
import io.realm.RealmObject
import io.realm.annotations.PrimaryKey

open class Day() : RealmObject() {

    @PrimaryKey
    var date: String = "" //d.M.yyyy
    var year = 0
    var month = 0
    var records = RealmList<Record>()

    constructor(date: String) : this() {
        this.date = date
        month = date.split(".")[1].toInt()
        year = date.split(".")[2].toInt()
    }
}
